pkg/collector/scheduler.go

```diff
@@ -134,10 +140,22 @@ func (s *CheckScheduler) getChecks(config integration.Config) ([]check.Check, er
 	checks := []check.Check{}
 	numLoaders := len(s.loaders)
 
+	initConfig := commonInitConfig{}
+	err := yaml.Unmarshal(config.InitConfig, &initConfig)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("Get checks for check '%s' with loader name '%s'", config.Name, initConfig.LoaderName)
+
 	for _, instance := range config.Instances {
 		errors := []string{}
 
 		for _, loader := range s.loaders {
+			// the loader is skipped if the loader name is set and does not match
+			if (initConfig.LoaderName != "") && (initConfig.LoaderName != loader.Name()) {
+				log.Debugf("Loader name %v does not match, skip loader %v for check %v", initConfig.LoaderName, loader.Name(), config.Name)
+				continue
+			}
```

Comment:
I think we can possibly deprecated `init_config.is_jmx` flag, since we can use `init_config.loader: jmx` instead. 
WDYT?

Not that `is_jmx` seems to be available as both init_config and instance level config.

Reply:
I think there's something a little weird currently. If we have `init_config.loader: jmx`, but there's no `is_jmx` setting, and the check isn't a default JMX check, then I think the JMX check would fail to be scheduled.  

Now, I don't think existing configs would break, because they wouldn't have the `loader` setting, but I feel like the jmx validation logic should probably take into account `init_config.loader: jmx` in the same way it currently looks at `is_jmx`.

And yes I believe, we'd have to support the per-instance `loader`.

Reply:
> Now, I don't think existing configs would break, because they wouldn't have the loader setting, but I feel like the jmx validation logic should probably take into account init_config.loader: jmx in the same way it currently looks at is_jmx.

Yes, agree, jmx validation logic needs to be updated to support `init_config.loader: jmx` and `instance[].loader: jmx`.

Will make those changes.

> And yes I believe, we'd have to support the per-instance loader.

👍

Reply:
@truthbk Update with:
- Make loader config available both at init_config and instance level
- Make jmx validation aware of loader config + tests